### PR TITLE
Remove Gulp from Project and Migrate to l10n Standard

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

Solves #2253 

As discussed in the past we wanted to move on from Gulp to reduce complexity in Zowe Explorer's package, allowing us to simplify builds and move onto any technologies in the future.

The removal of Gulp was done through the migration of our exisiting localization strategy with `i18n`, which is now replaced with suggested `l10n` from [VS Code](https://github.com/microsoft/vscode-l10n). This new `l10n` migration will allow easier localization accross our project(s) through a simpler way of compiling translation strings, as well as the ability to export now these localization files to XLF files as a potential option in the future.

### Additional documentation on `l10n` standard

- `l10n` repo https://github.com/microsoft/vscode-l10n
- Repo for tool used for creating localized strings from VS Code https://github.com/microsoft/vscode-l10n#vscodel10n-dev
- Sample repo used to migrate old `i18n` integration to `l10n` https://github.com/microsoft/vscode-extension-samples/tree/main/l10n-sample

## Release Notes

Milestone: `v3.0 pre-releases`

Changelog: Removed Gulp dependency and migrated from `i18n` standard to `l10n`.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by rudyflores** [comment]
FYI seems like @vscode/l10-dev is broken on node 20 so perhaps we should have our workflows run on node 18 until they resolve this.

link to issue: https://github.com/microsoft/vscode-l10n/issues/114

**Comment by rudyflores** [comment]
> FYI seems like @vscode/l10-dev is broken on node 20 so perhaps we should have our workflows run on node 18 until they resolve this.
> 
> link to issue: [microsoft/vscode-l10n#114](https://github.com/microsoft/vscode-l10n/issues/114)

Purposely added back command for exporting localization (can see the check break with same error from issue linked from dependency above) 

Perhaps we could use node 18 for these checks which in term should fix the issue showing in the checks? @t1m0thyj @zFernand0

**Review by zFernand0 — DISMISSED** [review]
changes LGTM! 😋

File: packages/zowe-explorer/src/webviews/src/edit-history/components/PersistentToolBar/PersistentClearAllButton.tsx [hunk 1]
```diff
@@ -13,9 +13,7 @@ import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
 import { JSXInternal } from "preact/src/jsx";
 import { useDataPanelContext } from "../PersistentUtils";
 import PersistentVSCodeAPI from "../PersistentVSCodeAPI";
-import * as nls from "vscode-nls";
-
-const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import * as nls from "@vscode/l10n";
```
[the remark this answers is not in the document]
Comment (zFernand0) [reply packages/zowe-explorer/src/webviews/src/edit-history/components/PersistentToolBar/PersistentClearAllButton.tsx:16]: no need to, but thanks for considering (and for the explanation too 😋)

**Comment by rudyflores** [comment]
I think we're ready to go ahead and review! Thanks for your help with the workflows @t1m0thyj  :)

**Review by traeok — APPROVED** [review]
LGTM, thanks @rudyflores! (and thanks for getting rid of gulp! 😋)

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**